### PR TITLE
Add initial delay for ActorSpawnManager

### LINKED ISSUE #16693: ActorSpawnManager should have an field for Initial delay
There is SpawnInterval but this cannot be used to control units to not get spawned at game start.
In a RA based mod where have implemented Ants as tickbox enabled map creeps, there will spawn one Ant right at start despite SpawnInterval: 5000 given, there should be an way to prevent this from occuring.

### LINKED ISSUE #20153: Delay first worm appearance on the map.
Worms start spawning right after game start, which really broke every early game. Loosing first harvester or delaying your eco in early game leave you very behind. But when worms start spawn after 10 or 15 min in game its not as big deal. (it dont affects eco so much as you have many harvesters at this point or you can use some cheap trike to get rid of worm)

Proposed solution

Implement new lobby option that allows set you delay for worm first appearance. This is already done in lot of custom maps via LUA script, but I thing it’s good idea to have this ability for all maps by default.

Alternative 

Hardcode spawn delay for first worm  somewhere between 8-15 minutes.

wormspawndelay

### PR DESCRIPTION
**Pull request by Porenutak**
closes #20153
closes #16693
part of #21154

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/ActorSpawnManager.cs [hunk 1]
```diff
@@ -80,6 +85,9 @@ protected override void Created(Actor self)
 
 		void ITick.Tick(Actor self)
 		{
+			if (--intialDelay > 0)
+				return;
+
```
Comment (PunkPun) [inline OpenRA.Mods.Common/Traits/World/ActorSpawnManager.cs:95]: you'll want to move this to bellow `IsTraitDisabled || !enabled)` and then add a reset of the cooldown on the trait being enabled.

```csharp
protected override void TraitEnabled(Actor self)
{
	intialDelay = info.InitDelay;
}
```

**Comment by PunkPun** [comment]
#20153 requires UI to be added does it not? I suppose adding a 7 minute default if good enough, just perhaps it should not affect missions?

**Comment by Porenutak** [comment]
That's beyound my coding skills (yet). This PR is quick fix, which I thing is good enough. Both missions and MP should be affected. You never want worms on map until mid game.

**Comment by Porenutak** [comment]
screen

@PunkPun you were right. According to D2k editor all missions have Initial worms sleep set to 50. After some testing its look like be +/- 5000 ticks. 
I update Missions intDelay to this value.

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/16fb2da72b9c4ee0af7d4c127043f4e710d0f9a3...7d8a5138be8d3ae6e2fe1abd5ed276dbcbeab630)

**Pull request merged**